app/workers/prune_event_logs_worker.rb

```diff
@@ -50,9 +50,18 @@ def perform
                             .where('created_at < ?', BACKLOG_DAYS.days.ago.beginning_of_day)
                             .where(event_type_id: event_type_ids)
 
+        # Keep the latest log per-resource for each day and event type (i.e. discard duplicates)
+        kept_logs = event_logs.distinct_on(:resource_id, :resource_type, :event_type_id, :created_date)
```

Comment:
Adding a composite index on these columns may help improve performance here. Now sure if the cost for writes is worth it, though, since we do have an index on each of these columns. We'll have to look at real-world performance after we deploy.